android/expoview/src/main/java/host/exp/exponent/notifications/schedulers/IntervalScheduler.java

```diff
@@ -0,0 +1,240 @@
+
+    ensureDetails();
+    try {
+      getManager().schedule(experienceId, notificationId, details, nextAppearanceTime, null);
+    } catch (ClassNotFoundException e) {
+      e.printStackTrace();
+    }
+  }
+
+  private void ensureDetails() {
+    try {
+      details = HashMapSerializer.deserialize(serializedDetails);
+    } catch (JSONException e) {
+      e.printStackTrace();
+    }
+  }
+
+  @Override
+  public void cancel() {
+    getManager().cancel(experienceId, notificationId);
+  }
+
+  @Override
+  public boolean canBeRescheduled() {
+    return repeat || (DateTime.now().toDate().getTime() < scheduledTime);
+  }
+
+  @Override
+  public void onPostSchedule() {
+    this.scheduled = true;
+    save();
+  }
+
+  @Override
+  public String saveAndGetId() {
+    details.put(SchedulersManagerProxy.SCHEDULER_ID, getIdAsString());
+    setDetails(details);
+    save();
+    return getIdAsString();
+  }
+
+  @Override
+  public String getOwnerExperienceId() {
+    return experienceId;
+  }
+
+  @Override
+  public String getIdAsString() {
+    return Integer.valueOf(id).toString();
+  }
+
+  @Override
+  public void remove() {
+    cancel();
+    delete();
+  }
+
+  private long getNextAppearanceTime() { // elapsedTime
+    // time when notification should be presented can be represented as (interval * t + scheduledTime)
+
+    long now = DateTime.now().toDate().getTime();
+    long whenShouldAppear = -1;
+    if (now <= scheduledTime) {
+      whenShouldAppear =  scheduledTime;
+    } else {
+
+      if (interval <= 0) {
+        throw new IllegalArgumentException();
+      }
+
+      now  = DateTime.now().toDate().getTime();
+      long elapsedTime = (now - scheduledTime);
+      long t = elapsedTime / interval + 1;
+      whenShouldAppear = interval * t + scheduledTime;
+    }
+
+    long bootTime = DateTime.now().toDate().getTime() - SystemClock.elapsedRealtime();
+    return whenShouldAppear-bootTime;
+  }
+
+  private ExponentNotificationManager getManager() {
+    return new ExponentNotificationManager(mApplicationContext);
+  }
+
+  // model getters and setters
+
+  public int getId() {
+    return id;
+  }
+
+  public void setId(int id) {
+    this.id = id;
+  }
+
+  public int getNotificationId() {
+    return notificationId;
+  }
+
+  public void setNotificationId(int notificationId) {
+    this.notificationId = notificationId;
+  }
+
+  public String getExperienceId() {
+    return experienceId;
+  }
+
+  public void setExperienceId(String experienceId) {
+    this.experienceId = experienceId;
+  }
+
+  public boolean isRepeat() {
+    return repeat;
+  }
+
+  public void setRepeat(boolean repeat) {
+    this.repeat = repeat;
+  }
+
+  public boolean isScheduled() {
+    return scheduled;
+  }
+
+  public void setScheduled(boolean scheduled) {
+    this.scheduled = scheduled;
+  }
+
+  public String getSerializedDetails() {
+    return serializedDetails;
+  }
+
+  public void setSerializedDetails(String serializedDetails) {
```

Comment:
<img width="675" alt="Zrzut ekranu 2019-05-20 o 10 14 56" src="https://user-images.githubusercontent.com/1151041/58006333-1f70d380-7ae8-11e9-8070-09dbae9875c1.png">

Even if so, when DBFlow uses `setSerializedDetails` to set serialized details, when you call `getDetails` you will get a null or old value since you don't update `details` when setting `serializedDetails` (this kind of bug, if it is a bug, would argue for using `Bundle` for options class)